tools/cordova/package-id-version-parser.js

```diff
@@ -0,0 +1,27 @@
+// Accepts a combined Cordova package ID + version string, then parses out
+// and returns the ID and version in a package details object.
+//
+// Example `packageIdAndVersion` formats:
+// some-cordova-plugin@1.0.0
+// @somescope/some-cordova-plugin@1.0.0
+exports.parse = packageIdAndVersion => {
+  const package = {};
+  if (packageIdAndVersion) {
+    const parts = packageIdAndVersion.match(
+      /^(@[^\/]*)?(\/)?([^@]*)(@)?(.*)?/
+    );
+    const details = {
+      scope: parts[1],
+      scopeSeparator: parts[2],
+      packageName: parts[3],
+      versionSeparator: parts[4],
+      version: parts[5],
+    };
```

Comment:
Maybe skip the `details` object?
```js
const [
  _matchText,
  scope,
  scopeSeparator,
  packageName,
  versionSeparator,
  version,
] = packageIdAndVersion.match(
  /^(@[^\/]*)?(\/)?([^@]*)(@)?(.*)?/
);
```

Reply:
@benjamn I actually did exactly that first, but was greeted with the following

```
TypeError: Cannot read property 'Symbol(Symbol.iterator)' of packageIdAndVersion.match
```

I decided to switch it around instead of chasing that down further, but I'll investigate. Thanks!

Reply:
And now for the life of me I can't re-create the same error. So, changes coming!